application/views/helpers/Lightgallery.php

```diff
@@ -0,0 +1,114 @@
+                            'controls' => true,
+                        ],
+                    ];
+                    if (isset($index['tracks'])) {
+                        foreach ($index['tracks'] as $key => $track) {
+                            $label = metadata($track, 'display_title');
+                            $srclang = (metadata($track, 'dcterms:language')) ? metadata($track, 'dcterms:language') : '';
+                            $type = (metadata($track, 'dcterms:type')) ? metadata($track, 'dcterms:type') : 'captions';
+                            $videoSrcObject['tracks'][$key]['src'] = $track->getWebPath();
+                            $videoSrcObject['tracks'][$key]['label'] = $label;
+                            $videoSrcObject['tracks'][$key]['srclang'] = $srclang;
+                            $videoSrcObject['tracks'][$key]['kind'] = $type;
+                        }
+                    }
+                    $videoSrcJson = json_encode($videoSrcObject);
+                    $videoThumbnail = ($media->hasThumbnail()) ? metadata($media, 'thumbnail_uri') : img('fallback-video.png');
+                    $html .= '<div data-video="' . html_escape($videoSrcJson) . '" ' . $mediaCaptionAttribute . 'data-thumb="' . html_escape($videoThumbnail) . '" data-download-url="' . $source . '" class="media resource">';
+                    $html .= '<img src="' . $videoThumbnail . '" alt="" role="presentation">';
+                } else if ($mediaType == 'application/pdf') {
+                    $html .= '<div data-iframe="' . html_escape($source) . '" '. $mediaCaptionAttribute . 'data-src="' . $source . '" data-thumb="' . html_escape(metadata($media, 'thumbnail_uri')) . '" data-download-url="' . $source . '" class="media resource">';
+                    $html .= file_markup($media);
+                } else {
+                    $html .=  '<div data-src="' . $source . '" ' . $mediaCaptionAttribute . 'data-thumb="' . html_escape(metadata($media, 'thumbnail_uri')) . '" data-download-url="' . $source . '" class="media resource">';
+                    $html .= file_markup($media);
+                }
+                $html .= '</div>';
+            }
+            $html .= '</div>';
+        } elseif (isset($sortedMedia['otherMedia'])) {
+            $otherMedia = $sortedMedia['otherMedia'];
+            $html .= '<div id="other-media" class="element">';
+            $html .= '<h3>' . get_view()->translate('Other Media') . '</h3>';
+            foreach($otherMedia as $media) {
+                $html .= '<div class="element-text">' . link_to($media, null, metadata($media, 'display_title')) . '</div>';
+            }
+            $html .= '</div>';
+        }
+
+        return $html;
+    }
+
+    protected function _prepareLightgalleryFiles($files = null) {
+        $sortedMedia = [];
+        $whitelist = ['image/bmp', 'image/gif', 'image/jpeg', 'image/png', 'image/svg+xml', 'video/flv', 'video/x-flv', 'video/mp4', 'video/m4v',
```

Comment:
This ultimately sources in part from the HTML5 Media plugin, right? Not sure that the flv ones will really work here (or actually that they still work in HTML5 Media in the absence of Flash). Have you tested it?

Similarly WMV, though maybe Windows machines can play that through a video tag? And I'd actually expect it to be taking Ogg video also, though realistically MP4 under its various names and to a lesser extent WebM are all that are relevant. For reference the core renders as HTML5 video tags the following types:

`video/mp4`, `video/x-m4v`,  `video/ogg`, `video/webm`, `video/quicktime`